### PR TITLE
Feature/add player vent event

### PR DESCRIPTION
**Pull request by Hiumee**
Addressing issue #148

Added event
+ `IPlayerVentEvent : IPlayerEvent`
  + `VentId : VentLocation`
  + `VentEnter : bool`

Added enum
+ `VentLocation : uint`

### CODE REVIEW

**Comment by AeonLucid** [comment]
Looks pretty good to me and I am okay with merging this, just not sure about the `VentLocation` enum. 
Wondering if anyone has ideas / thoughts about this before merging.

File: src/Impostor.Api/Innersloth/VentLocation.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        SkeldWeapons = 7,
+        SkeldLowerReactor = 8,
+        SkeldLowerEngine = 9,
+        SkeldShields = 10,
+        SkeldUpperReactor = 11,
+        SkeldUpperNavigation = 12,
+        SkeldLowerNavigation = 13,
+
+        // Mira HQ
+        MiraBalcony = 1,
```
Comment (gavinsteinhoff) [inline src/Impostor.Api/Innersloth/VentLocation.cs:22]: Why does this one start at 1?

File: src/Impostor.Api/Innersloth/VentLocation.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+namespace Impostor.Api.Innersloth
+{
+    public enum VentLocation : uint
```
Comment (gavinsteinhoff) [inline src/Impostor.Api/Innersloth/VentLocation.cs:3]: Should these be separate enums? How does this work with them all being in the same enum?

Comment (AeonLucid) [reply src/Impostor.Api/Innersloth/VentLocation.cs:3]: I think this is fine. If you write plugin code you can easily do 

```csharp
if (location == VentLocation.SkeldMedbay ||
    location == VentLocation.MiraMedbay ||
    location == VentLocation.PolusLaboratory) {

}
```

File: src/Impostor.Api/Innersloth/VentLocation.cs (hunk 1 above)
Comment (Hiumee) [reply src/Impostor.Api/Innersloth/VentLocation.cs:22]: > Why does this one start at 1?

Got the ids from here
https://github.com/codyphobe/among-us-protocol/blob/master/07_miscellaneous/04_map_specific_ids_for_vents_and_tasks.md

**Comment by js6pak** [comment]
I feel like enum should be replace with something struct-like in Impostor.Api.Innersloth so it can include position - which would be required for anticheat

File: src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ public InnerPlayerControl(ILogger<InnerPlayerControl> logger, IServiceProvider s
             _eventManager = eventManager;
             _game = game;
 
-            Physics = ActivatorUtilities.CreateInstance<InnerPlayerPhysics>(serviceProvider, this);
+            Physics = ActivatorUtilities.CreateInstance<InnerPlayerPhysics>(serviceProvider, this, _eventManager, _game);
```
Comment (Mr-VIT) [inline src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs:29]: https://github.com/Impostor/Impostor/pull/197#discussion_r532611352
> Why is _eventManager here? It is in the serviceProvider.

**Pull request merged**